packages/theme-variants/variants.js

```diff
@@ -170,14 +171,16 @@ export default class Variants {
       return;
     }
 
-    var newurl =
-      window.location.protocol +
-      '//' +
-      window.location.host +
-      window.location.pathname +
-      '?variant=' +
-      variant.id;
-    window.history.replaceState({ path: newurl }, '', newurl);
+    var currentHref = window.location.href
+    var newUrl = ''
+
+    if (/variant=/.test(currentHref)) {
+      newUrl = currentHref.replace(/(variant=)[^\&]+/, '$1' + variant.id)
```

Comment:
The `$1` is represented as the regex match, which is `variant=`. And we want to keep `variant=` in our URL and append it with the new `variant.id`, therefore using of `$1` is needed.

Reply:
I also considered using the URLSearchParameters with the polyfill. Since it added an extra dependency to the library, I didn't think it was necessary to do that.